gladier/state_models.py

```diff
@@ -268,3 +253,49 @@ def result_path_for_step(self) -> str:
         )
         result_path = ensure_json_path(result_path)
         return result_path
+
+
+class BaseCompositeState(BaseState):
+    state_type: str = "CompositeVirtualState"
+    state_name_prefix: str = ""
+
+    @abstractmethod
+    def construct_flow(self) -> BaseState:
+        raise ValueError(
+            f"construct_flow method not implemented on class {type(self).__name__}"
+        )
+
+    def get_flow_definition(self) -> JSONObject:
+        start_state = self.construct_flow()
+        flow_definition = start_state.get_flow_definition()
+        if self.state_name_prefix:
+            new_states: JSONObject = {}
+            flow_states: JSONObject = flow_definition["States"]
+            for state_name, state_def in flow_states.items():
+                new_states[self.state_name_prefix + state_name] = copy.deepcopy(
+                    state_def
+                )
+            for new_state_name, new_state_def in new_states.items():
+                for state_def_key, state_def_val in new_state_def.items():
+                    # Replace all references to old state names with prefixed value
+                    if state_def_val in flow_states.keys():
+                        new_states[new_state_name][state_def_key] = (
+                            self.state_name_prefix + state_def_val
+                        )
+            flow_definition["States"] = new_states
+
+        return flow_definition
```

Comment:
The state renaming code might be nice to have in its own separate method for clarity if people are running `super().get_flow_definition()` a lot. I'm not sure if we'd ever want to call only the state naming functionality out-of-band, but it could possibly be useful. Maybe call it `generate_state_names()`?